master/internal/api_user.go

```diff
@@ -683,3 +683,116 @@ func (a *apiServer) PostUserActivity(
 	}
 	return &apiv1.PostUserActivityResponse{}, err
 }
+
+func (a *apiServer) PostLongLivedToken(
+	ctx context.Context, req *apiv1.PostLongLivedTokenRequest,
+) (*apiv1.PostLongLivedTokenResponse, error) {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if err = user.AuthZProvider.Get().CanCreateUsersOwnToken(
+		ctx, *curUser); err != nil {
+		return nil, status.Error(codes.PermissionDenied, err.Error())
+	}
+
+	tokenExpiration := time.Now().Add(user.TokenExpirationDuration)
```

Comment:
Is the expectation here to be handling things in local time? It might be a good opportunity to handle this in UTC & allow the frontend side of things do the conversion to local time

Reply:
Yes, I thought the same. Added the time as UTC in the table creation but forgot to factor in UTC time here. Thanks for catching this.

Reply:
Updated in a manner to have current UTC time call made only during insertion of the <expires_at> and <created_at>. Let me know if it does make sense